include/cantera/oneD/StFlow.h

```diff
@@ -325,25 +312,144 @@ class StFlow : public Domain1D
 
+    /**
+     * Evaluate the lambda equation residual.
+     *
+     * The function calculates the lambda equation as
+     * @f[
+     *    \frac{d\Lambda}{dz} = 0
+     * @f]
+     *
+     * The lambda equation serves as an eigenvalue that allows the momentum
+     * equation and continuity equations to be simultaneously satisfied in
+     * axisymmetric flows. The lambda equation propgates information from
+     * left-to-right. The default boundary condition is zero (@f$ \Lambda @f$)
+     * at the left and zero flux at the right boundary.
+     */
+    virtual void evalLambda(double* x, double* rsd, int* diag,
+                            double rdt, size_t jmin, size_t jmax);
+
+    /**
+     * Evaluate the energy equation residual.
+     *
+     * The function calculates the energy equation:
+     * @f[
+     *    \rho c_p u \frac{dT}{dz} =
+     *    \frac{d(k \frac{dT}{dz})}{dz} - \sum_k (\omega_k h_{k_{\text{ref}}}) -
+     *    \sum_k \left( \frac{J_k c_{p_k}}{M_k} \frac{dT}{dz} \right)
+     * @f]
+     *
+     * The energy equation includes contributions from
+     * chemical reactions and diffusion. Default is zero temperature (@f$ T @f$)
+     * at the left and right boundaries. These boundary values are updated by the
+     * specific boundary object connected to the domain.
+     */
+    virtual void evalEnergy(double* x, double* rsd, int* diag,
+                            double rdt, size_t jmin, size_t jmax);
+
+    /**
+     * Evaluate the species equations' residuals.
+     *
+     * The function calculates the species equations as
+     * @f[
+     *    \rho u \frac{dY_k}{dz} + \frac{dJ_k}{dz} = M_k\omega_k
+     * @f]
+     *
+     * The species equations include terms for temporal and spatial variations
+     * of species mass fractions (@f$ Y_k @f$). The default boundary condition is zero
+     * flux for species at the left and right boundary.
+     */
+    virtual void evalSpecies(double* x, double* rsd, int* diag,
+                             double rdt, size_t jmin, size_t jmax);
+
+    /**
+     * Evaluate the electric field equation residual to be zero everywhere.
+     *
+     * The electric field equation is implemnted in the IonFlow class. The default
+     * boundary condition is zero electric field (@f$ E @f$) at the boundary,
+     * and @f$ E @f$ is zero within the domain.
+     */
+    virtual void evalElectricField(double* x, double* rsd, int* diag,
+                                   double rdt, size_t jmin, size_t jmax);
+
     /**
      * Update the thermodynamic properties from point j0 to point j1
      * (inclusive), based on solution x.
+     *
+     * The gas state is set to be consistent with the solution at the
+     * points from j0 to j1.
+     *
+     * Properties that are computed and cached are:
+     * - m_rho (density)
+     * - m_wtm (mean molecular weight)
+     * - m_cp (specific heat capacity)
+     * - m_hk (species specific enthalpies)
+     * - m_wdot (species production rates)
```

Comment:
To generate links to member variables in Doxygen:
```suggestion
     * - #m_rho (density)
     * - #m_wtm (mean molecular weight)
     * - #m_cp (specific heat capacity)
     * - #m_hk (species specific enthalpies)
     * - #m_wdot (species production rates)
```